src/iceberg_manifest.cpp

```diff
@@ -221,6 +223,11 @@ void IcebergManifestEntryV2::PopulateNameMapping(idx_t column_id, const LogicalT
 
 AvroScan::AvroScan(const string &scan_name, ClientContext &context, const string &path) : context(context) {
 	auto &instance = DatabaseInstance::GetDatabase(context);
+	if (!__AVRO_LOADED__) {
+		ExtensionHelper::AutoLoadExtension(instance, "avro");
+		__AVRO_LOADED__ = true;
+	}
+
 	auto &avro_scan_entry = ExtensionUtil::GetTableFunction(instance, "read_avro");
```

Comment:
nit: perhaps we want to wrap extension loading and `GetTableFunction` into a single helper function. Otherwise, I can see this problem reoccurring the next time `read_avro` is used in a different path

Reply:
Normally we like to load all necessary extensions in `LoadInternal` in `iceberg_extension.cpp` so that avro loading happens during the iceberg extension load. The problem is that in our CI, we test compiling/linking all the extensions together to verify behavior between extensions. This uncovered a problem where Avro has conflicts with some of our other extensions, so we don't link it. Not linking the extension means we can't auto load it in our tests (but `require avro` still works). Since avro cannot be auto-loaded, all tests that auto-load iceberg fail if avro gets autoloaded. The temporary solution for now is to only load avro when the avro scan function is used. In the future we plan to link the avro extension so we can load it when the iceberg extension gets loaded